src/main/kotlin/com/atkinsondev/opentelemetry/build/OpenTelemetryBuildPlugin.kt

```diff
@@ -89,7 +89,14 @@ class OpenTelemetryBuildPlugin : Plugin<Project> {
                         val buildListener = OpenTelemetryBuildListener(rootSpan, openTelemetry, project.logger)
                         project.gradle.addBuildListener(buildListener)
 
-                        val taskListener = OpenTelemetryTaskListener(tracer, rootSpan, baggage, project.logger)
+                        val taskListener =
+                            OpenTelemetryTaskListener(
+                                tracer = tracer,
+                                rootSpan = rootSpan,
+                                baggage = baggage,
+                                logger = project.logger,
+                                hierarchicalTestSpans = extension.hierarchicalTestSpans.get(),
+                            )
```

Comment:
At first I wanted to pass in the whole `extension` into the listener. 

(Even though I already thought that one could argue that this would be a unnecessary tight coupling between the two classes).

I went with just `hierarchicalTestSpans` because of the tests as otherwise I'd have had to mock the extension somehow.

Reply:
+1 to only passing in the test-spans flag to avoid tightly coupling the extension class to the listener, as you mentioned.